File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/detail/Log.java
package io.github.orioncraftmc.meditate.internal.detail;

//struct YGNode;
//struct YGConfig;


import io.github.orioncraftmc.meditate.internal.YGConfig;
import io.github.orioncraftmc.meditate.internal.YGNode;
import io.github.orioncraftmc.meditate.internal.enums.YGLogLevel;
import org.jetbrains.annotations.Nullable;

public class Log {

    public static void log(@Nullable YGNode node, YGLogLevel level, Object context, String format, Object... args) {
        GlobalMembers.vlog(node == null ? null : node.getConfig(), node, level, context, format, args);
    }

    public static void log(YGConfig config, YGLogLevel level, Object context, String format, Object... args) {
        GlobalMembers.vlog(config, null, level, context, format, args);
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/detail/Log.kt
package app.cash.redwood.yoga.detail

import app.cash.redwood.yoga.YGConfig
import app.cash.redwood.yoga.YGNode
import app.cash.redwood.yoga.enums.YGLogLevel

object Log {
  fun log(
    node: YGNode?,
    level: YGLogLevel,
    context: Any?,
    format: String,
    vararg args: Any?,
  ) {
    GlobalMembers.vlog(node?.getConfig(), node, level, context, format, *args)
  }

  fun log(
    config: YGConfig?,
    level: YGLogLevel,
    context: Any?,
    format: String,
    vararg args: Any?,
  ) {
    GlobalMembers.vlog(config, null, level, context, format, *args)
  }
}
